hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -342,13 +356,14 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	})
 
 	// Validate platform specific input.
-	var ami string
-	if nodePool.Spec.Platform.Type == hyperv1.AWSPlatform {
+	rhcosInfo := &rhcosDetails{}
```

Comment:
@enxebre I guess this would be another interesting point where potentially every provider except `none` would have to diverge regaring to what exactly to render in the machine template. Did you already think about a structure on how you want this to look like?

I just hacked something in here at the moment.